### ISSUE TITLE
The problem frameworks are trying to solve

### ISSUE THREAD

**Issue by eliasmalik** [foundersandcoders/mc-react-solves-what-now#2]
> This is the problem that the majority of front end frame works are trying to solve. 

Front-end frameworks are trying to solve the problem of managing large and complex UIs built on-top of technologies that were originally conceived for very different paradigms.

The problem in the jQuery days is that many applications used the DOM as their source of the canonical state of the UI. But many aspects of application state aren't possibly or easy to store in the DOM, or require convoluted calculations based on HTML attribute values and the like, so often people _also_ stored app state in JS objects. This mixed approach leads to codebases that are difficult to manage.

Broadly, React's approach to this problem is:
* Canonical application state is stored entirely in JS
* Treat the DOM explicitly as a low-level presentation layer. Interact with it only through an abstraction
* Operate on the level of higher-level component-style abstractions which represent subtrees of the DOM

> In this morning challenge you will try to dynamically update the DOM as state changes.

In your challenge, app state is already stored entirely in JS, so you've already adopted one of React's approaches to solving the problems outlined above. Was this the intention?

**Comment by oliverjam** [comment]
>In your challenge, app state is already stored entirely in JS, so you've already adopted one of React's approaches to solving the problems outlined above. Was this the intention?

Yeah the idea was to have them think about the problem of having their UI automatically re-render when the JS state object changes. Basically we wanted them to write a `setState` function that updated the global state object and then destroyed the DOM tree and rebuilt it from the new state.

We also encouraged them to not do direct DOM manipulation (someone solved the problem very quickly by just changing the buttons `innerText`, but we tried to show them why this would be a problem in a larger codebase)

**Comment by astroash** [comment]
@eliascodes / @oliverjam are either of you able to type up a more appropriate intro to this morning challenge? The current version is something I mashed together during the mc 😅

**Comment by sofiapoh** [comment]
If either of you don't write it I can since I wrote it originally! 😄
